brooklyn-server/core/src/main/java/org/apache/brooklyn/core/entity/AbstractApplication.java

```diff
@@ -220,13 +221,34 @@ public void stop() {
                 //but that could be handled by the impl at management
                 //(keeping recently unmanaged things)  
                 //  however unmanaging must be done last, _after_ we stop children and set attributes 
+                Location location = Iterables.getFirst(getLocations(), null);
```

Comment:
Could getLocations return an empty Collection? If so, we will use the default value of null and throw a null pointer exception in unmanageProvisionerIfNotUsed